ankihub/labs/llm/llm.py

```diff
@@ -0,0 +1,482 @@
+            ]
+            yaml_files.sort()
+            return yaml_files or ["No Anki templates found"]
+        except Exception:
+            return ["Error listing templates"]
+
+
+class PromptPreviewDialog(QDialog):
+    """Dialog for previewing and editing a prompt template before execution."""
+
+    def __init__(self, parent, template_name: str, editor: Editor) -> None:
+        super().__init__(parent)
+        self.template_name = template_name
+        self.editor = editor
+        self.template_content = TemplateManager.get_template_content(template_name)
+        self._setup_ui()
+
+    def _setup_ui(self) -> None:
+        """Set up the dialog UI."""
+        self.setWindowTitle(f"Preview Template: {self.template_name}")
+        self.setMinimumWidth(800)
+        self.setMinimumHeight(600)
+
+        layout = QVBoxLayout()
+
+        # Add description label
+        description = QLabel("Review and edit the prompt template below:")
+        description.setWordWrap(True)
+        layout.addWidget(description)
+
+        # Add template editor
+        self.template_edit = QTextEdit()
+        self.template_edit.setPlainText(self.template_content)
+        self.template_edit.setFont(QFont("Consolas"))
+        layout.addWidget(self.template_edit)
+
+        # Add button row
+        button_layout = QHBoxLayout()
+
+        # Execute button
+        execute_button = QPushButton("Execute Prompt")
+        execute_button.clicked.connect(self._on_execute)
+        button_layout.addWidget(execute_button)
+
+        # Cancel button
+        cancel_button = QPushButton("Cancel")
+        cancel_button.clicked.connect(self.reject)
+        button_layout.addWidget(cancel_button)
+
+        layout.addLayout(button_layout)
+        self.setLayout(layout)
+
+    def _on_execute(self) -> None:
+        """Handle the execute button click."""
+        modified_template = self.template_edit.toPlainText()
+        # TODO Save the modified template if it differs from the original
+        _execute_prompt_template(self.editor, self.template_name, modified_template)
+        self.accept()
+
+
+def _check_and_install_uv() -> None:
+    """Check if uv is installed and install it if not."""
+    try:
+        subprocess.run(["uv", "version"], capture_output=True, check=True)
+        print("uv is installed")
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        try:
+            if platform.system() == "Darwin":  # macOS
+                subprocess.run(
+                    "curl -LsSf https://astral.sh/uv/install.sh | sh",
+                    shell=True,
+                    check=True,
+                )
+            elif platform.system() == "Windows":
+                subprocess.run(
+                    'powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"',
+                    shell=True,
+                    check=True,
+                )
```

Comment:
:wrench: To make it support Linux:
```suggestion
            if platform.system() == "Windows":
                subprocess.run(
                    'powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"',
                    shell=True,
                    check=True,
                )
            else:  # macOS and Linux
                subprocess.run(
                    "curl -LsSf https://astral.sh/uv/install.sh | sh",
                    shell=True,
                    check=True,
                )
```